packages/react-components/src/menu/src/MenuItem.tsx

```diff
@@ -1,31 +1,44 @@
 import { Box } from "../../box";
-import { Keys, cssModule, mergeProps, useEventCallback, useSlots } from "../../shared";
+import { ElementType, ForwardedRef, KeyboardEvent, ReactElement, ReactNode, useMemo } from "react";
+import { InteractionStatesProps, Keys, cssModule, forwardRef, mergeProps, useEventCallback, useSlots } from "../../shared";
 import { SelectionMode } from "./selectionMode";
 import { Text } from "../../text";
-import { TooltipTrigger } from "../../tooltip";
-import { any, bool, elementType, func, object, oneOfType, string } from "prop-types";
-import { forwardRef, useMemo } from "react";
+import { TooltipTrigger, TooltipTriggerProps } from "../../tooltip";
 import { isNil } from "lodash";
 import { useMenuContext } from "./MenuContext";
 
-const propTypes = {
+export interface InnerMenuItemProps extends InteractionStatesProps {
+    /**
+     * @ignore
+     */
+    id?: string;
     /**
      * Matching collection item.
      */
-    item: object.isRequired,
+    item: {
```

Comment:
probably won't look good in the documentation

Reply:
the type is `{ key: string; tooltip?: { props: Record<string, any>; content: ReactElement<any, string | JSXElementConstructor<any>>; }; }`

is this acceptable for this case? or we prefer to remove the type validation and go with "object"

Reply:
For Listbox you used `CollectionItem` how does it look?

Reply:
![image](https://user-images.githubusercontent.com/38871812/114090042-e7faa400-9884-11eb-9b91-c3b0df5968d9.png)

Do we prefer to use a type for documentation like object, but not having the correct type validation, or do we prefer to keep CollectionItem and have any in the documentation?